### PR TITLE
Beautify plat-stm

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: core/arch/arm/plat-stm/platform_config.h [hunk 1]
```diff
@@ -130,30 +165,17 @@
...
-#define DEVICE2_VA_BASE		DEVICE2_PA_BASE
-#define DEVICE2_SIZE		CORE_MMU_DEVICE_SIZE
-#define DEVICE2_TYPE		MEM_AREA_IO_SEC
-
-#endif /*PLATFORM_CONFIG_H*/
+/* Make stacks aligned to data cache line length */
+#define STACK_ALIGNMENT		32
+
+/* default heap size */
+#define HEAP_SIZE		(24 * 1024)
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-stm/platform_config.h]: Is `CFG_CORE_HEAP_SIZE` too large so it has to be overridden?

File: core/arch/arm/plat-stm/rng_support.c [hunk 2]
```diff
@@ -62,27 +52,17 @@ static vaddr_t rng_base(void)
...
 	do {
 		status = read32(rng_base() + RNG_STATUS_OFFSET);
 	} while ((status & 0x20) != 0x20);
 
-	/* Check STATUS (see rng_fspec_revG_120720.pdf) */
-	if ((status & 0x3) != 0) {
-		EMSG("generated HW random data are not valid");
-		res = -1;
-	}
+	if (status & 0x3)
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-stm/rng_support.c]: `0x3` is a magic number.

File: core/arch/arm/plat-stm/rng_support.c [hunk 3]
```diff
@@ -62,27 +52,17 @@ static vaddr_t rng_base(void)
...
+		panic();
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-stm/rng_support.c]: Is panic the most useful behavior here?

File: core/arch/arm/plat-stm/main.c [hunk 4]
```diff
@@ -115,3 +131,132 @@ vaddr_t pl310_base(void)
...
+	 * Shared attribute internally ignored (bit22=1, bit13=0)
+	 * Parity disabled (bit21=0)
+	 * Event monitor disabled (bit20=0)
+	 * Set or preserved way config: size (bit19:17), ass (bit16)
+	 * Store buffer device limitation enabled (bit11=1)
+	 * Cacheable accesses have high prio (bit10=0)
+	 * Full Line Zero (FLZ) disabled (bit0=0)
+	 */
+#ifdef PL310_PREVERSE_AUXCTRL_WAY_CONFIG
+#define PL310_WAYCFG_BITMASK	SHIFT_U32(0xF, 16)
```
Comment (etienne-lms) [inline core/arch/arm/plat-stm/main.c]: ohhh! i didn't catch it. anyway, I'll drop it. I will introduce it back if we get a large scale of soc variants we would like vanilla op-tee to support. In the mean time, we'll go for new flavors or local patches.

**Comment by etienne-lms** [comment]
a rebase is needed (conflit with SCR fix merged).

Many beautification could apply to plat-imx. I think I should rather propose the changes to the maintainers rather than pushing straight a PR.

plat-imx-mx6qsabresd should be fixed against `cpu_gic_init()` for secondary cores. I should open an Issue ? (i have no imx6/sabre).

The generic gic interrupt scan fails on my stm (b2260). it detects to few active interrupts lines and I must pre-configure more lines (actually I configure all of them) as non-secure to get linux to boot.

File: core/arch/arm/plat-stm/main.c [hunk 5]
```diff
@@ -115,3 +131,132 @@ vaddr_t pl310_base(void)
...
+	assert(!cpu_mmu_enabled());
+
+	/* Allow NSec to manage FIQ/Imprecise abort (SCR[FW]=1, SCR[AW]=1) */
+	write_scr(SCR_AW | SCR_FW);
+
+	if (get_core_pos())
+		return;
+
+	/* both secure CPU access SCU */
+	write32(3, SCU_BASE + SCU_SAC);
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-stm/main.c]: Not updated

File: core/arch/arm/plat-stm/main.c [hunk 6]
```diff
@@ -115,3 +131,132 @@ vaddr_t pl310_base(void)
...
+
+	/* both nonsec cpu access SCU, private and global timer */
+	write32(0x333, SCU_BASE + SCU_NSAC);
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-stm/main.c]: Not updated

File: core/arch/arm/plat-stm/main.c [hunk 7]
```diff
@@ -115,3 +131,132 @@ vaddr_t pl310_base(void)
...
+
+#ifndef SCU_PREVERSE_CPU_PORT_FILT
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-stm/main.c]: Not updated

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
many thanks for the deep review and help.
i'll rebase, pack and tag.

**Comment by etienne-lms** [comment]
reviewed-by are missing in the 2 last commit.
i push back.

**Comment by jforissier** [comment]
About the checkpatch warnings:
- "externs should be avoided in .c files": I think the functions `arm_cl2_config()` and `plat_cpu_reset_late()` should be declared in a `.h` file since they are now implemented in C in `plat-stm`. What do you think?
- please add a newline in `tz_a9init.S` as requested
-  the error on `arm32.h` (inline assembly) is obviously a false positive, OK to ignore it

**Comment by etienne-lms** [comment]
sorry, i did not run checkpatch manually.

> "externs should be avoided in .c files": I think the functions arm_cl2_config() and plat_cpu_reset_late() should be declared in a .h file since they are now implemented in C in plat-stm. What do you think?

actaully there are already APIs exported in the core. ok i will declare them in a header.

> * the error on arm32.h (inline assembly) is obviously a false positive, OK to ignore it

right

**Comment by jforissier** [comment]
Thanks for the updates.
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by etienne-lms** [comment]
rebased, tags applied.

**Pull request merged**